### ISSUE TITLE
Inner grid width not recalculated after hiding columns in dynamic mode

### ISSUE THREAD

**Issue by jamesonhill** [Autodesk/react-base-table#177]
tldr if you are using dynamic row heights and remove/hide a column, the inner grid width is not recalculated.

Repro steps:
1) clone this repo (https://github.com/jamesonhill/react-base-table-repro) and run `yarn && yarn start` 
2) Click the button to remove the first column
3) Horizontally scroll to the end of the grid

Expected:
1) Inner grid width is calculated and displays as total columns width

Actual:
1) Inner grid width still includes the width of the hidden column

This only seems to happen in dynamic mode. You can technically work around it by doing `        tableRef.current?.table?.bodyRef.resetAfterColumnIndex(0, false);`, however that's pretty hacky and should work out of the box.

**Comment by jamesonhill** [comment]
Doing something like this seems to fix it: 
` componentDidUpdate(prevProps, prevState) {
    if (
      this.props.columns.some(col => {
        const prevcol = prevProps.columns.find(x => x.key === col.key);
        return prevcol.hidden !== col.hidden;
      })
    ) {
      this.table && this.table.resetAfterColumnIndex(0, false);
       this.leftTable && this.leftTable.resetAfterColumnIndex(0, false);
      this.rightTable && this.rightTable.resetAfterColumnIndex(0, false);
    }`

**Comment by jamesonhill** [comment]
I'm wondering if there is a better way to fix it though.

**Comment by nihgwu** [comment]
Thanks for your detailed information, yes there is a better way to fix it, but there are another issues(the rowHeight is not re-measured with the new columsn) with the same cause, so I'll create a PR to solve them, BTW you can try to use the playground to share the repro demo :smile:

**Comment by jamesonhill** [comment]
Oh, nice I didn't know you could use shareable links on the playground. I'll add that example, one minute.

**Comment by nihgwu** [comment]
no need, I can repro it locally, and I'm working on a fix on that
